pootle/apps/pootle_store/models.py

```diff
@@ -1753,6 +1781,13 @@ def update(self, overwrite=False, store=None, only_newer=False):
         self.state = LOCKED
         self.save()
 
+        store_revision = None
+        if hasattr(store, "parseheader"):
+            pootle_revision = store.parseheader().get("X-Pootle-Revision",
+                                                      None)
+            if pootle_revision:
+                store_revision = int(pootle_revision)
```

Comment:
This is the second appearance of this block of code. Please refactor to a method.